### ISSUE TITLE
DL3041 is being triggered when adding dnf parameters

### ISSUE THREAD

**Issue by VictorGFM** [hadolint/hadolint#945]
Even though the `dnf install -y <package>-<version>` format is already specified, the use of extra dnf parameters is causing the `DL3041` warning to be triggered.

- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
DL3041 should not be triggered when adding dnf parameters.

### Actual behavior
DL3041 is being triggered when adding dnf parameters.

### Steps to reproduce the behavior
Run `docker run --rm -i hadolint/hadolint:v2.12.1-beta < Dockerfile`.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
-:42 DL3041 warning: Specify version with `dnf install -y <package>-<version>`.
```

**Dockerfile** (if relevant)

```Dockerfile
RUN curl -LO https://dev.mysql.com/get/mysql80-community-release-el8-4.noarch.rpm && \
  rpm -i mysql80-community-release-el8-4.noarch.rpm && \
  mkdir -p /mnt/rootfs && \
  dnf --installroot /mnt/rootfs install curl-7.76.1-19.el9_1.1 mysql-community-client-8.0.32-1.el8 \
  -y --releasever 9 --setopt install_weak_deps=false --nodocs && \
  dnf --installroot /mnt/rootfs clean all && \
  rm -rf mysql80-community-release-el8-4.noarch.rpm
```

**Additional environment details (OS, stack version, etc.)**

**Comment by ivanov17** [comment]
I can't reproduce this bug with `microdnf` command (it's already supported as I see in the code), but I noticed strange behavior of this rule when I install packages multiple times in the same bash pipeline.

In the first case, the rule is triggered because the `tini` package does not have a version:
```
...
    && microdnf --assumeyes install epel-release epel-next-release \
    && microdnf --assumeyes --nodocs --setopt=install_weak_deps=0 \
    install rspamd-3.* tini && microdnf --assumeyes clean all \
...
```

 In the second case, the `tini` package has a version, and the rule does not trigger. 
```
...
    && microdnf --assumeyes install epel-release epel-next-release \
    && microdnf --assumeyes --nodocs --setopt=install_weak_deps=0 \
    install rspamd-3.* tini-0.* && microdnf --assumeyes clean all \
...
```

But in both cases, packages `epel-release` and `epel-next-release` do not have versions and with this the rule does not work at all.

Also, if the rule requires having versions not for one, but for all specified packages, then this is stupid. Only one application runs in a container. As for other packages, this is the job of the package manager.

Nothing like this is mentioned in the Docker best practices guide:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

And even if the rule is supposed to work this way, why does it ignore some packages?

Unfortunately, Hadolint is not yet ready for production.

**Comment by rndmh3ro** [comment]
The problem from @VictorGFM is that they're missing some `=` in the `RUN`-command:

This doesn't work:

```
RUN curl -LO https://dev.mysql.com/get/mysql80-community-release-el8-4.noarch.rpm && \
  rpm -i mysql80-community-release-el8-4.noarch.rpm && \
  mkdir -p /mnt/rootfs && \
  dnf --installroot /mnt/rootfs install curl-7.76.1-19.el9_1.1 mysql-community-client-8.0.32-1.el8 \
  -y --releasever 9 --setopt install_weak_deps=false --nodocs && \
  dnf --installroot /mnt/rootfs clean all && \
  rm -rf mysql80-community-release-el8-4.noarch.rpm
```

This works:

```
RUN curl -LO https://dev.mysql.com/get/mysql80-community-release-el8-4.noarch.rpm && \
  rpm -i mysql80-community-release-el8-4.noarch.rpm && \
  mkdir -p /mnt/rootfs && \
  dnf --installroot=/mnt/rootfs install curl-7.76.1-19.el9_1.1 mysql-community-client-8.0.32-1.el8 \
  -y --releasever=9 --setopt=install_weak_deps=false --nodocs && \
  dnf --installroot=/mnt/rootfs clean all && \
  rm -rf mysql80-community-release-el8-4.noarch.rpm
```

I don't know if hadolint is entirely at fault here. While using the arguments without `=`works, the docs of microdnf state that you should use them:

```
[root@dda08c804350 root]# microdnf --help
Usage:
  microdnf [OPTION…] COMMAND

[...]
Application Options:
  --assumeno                    Automatically answer no for all questions
  -y, --assumeyes               Automatically answer yes for all questions
  --best                        Try the best available package versions in transactions
  --config=<config file>        Configuration file location
  --disablerepo=ID              Disable repository by an id
  --disableplugin=name          Disable plugins by name
  --enablerepo=ID               Enable repository by an id
  --enableplugin=name           Enable plugins by name
  --nobest                      Do not limit the transaction to the best candidates
  --installroot=PATH            Set install root
  --nodocs                      Install packages without docs
  --noplugins                   Disable loading of plugins
  --refresh                     Set metadata as expired before running the command
  --releasever=RELEASEVER       Override the value of $releasever in config and repo files
  --setopt=<option>=<value>     Override a configuration option (install_weak_deps=0/1, allow_vendor_change=0/1, keepcache=0/1, module_platform_id=<name:stream>, cachedir=<path>, reposdir=<path1>,<path2>,..., tsflags=nodocs/test, varsdir=<path1>,<path2>,..., repo_id.opt
ion_name=<value>)
```

**Comment by sparrowt** [comment]
Regarding this bit:
> But in both cases, packages epel-release and epel-next-release do not have versions and the rule does not work for them at all.

that may be https://github.com/hadolint/hadolint/issues/1002 which seems like packages with a `-` in the name are not detected even if unpinned
